### ISSUE TITLE
Feature Request: Equivalent to React Native's "hitSlop"

### ISSUE THREAD

**Issue by zzzev** [flutter/flutter#14794]
React Native has a [very useful property that devs can set on Views called "hitSlop"](https://facebook.github.io/react-native/docs/view.html#hitslop) which increases the tappable size of an element without changing the way it interacts with layout.

The RN version of this has some limitations ("The touch area never extends past the parent view bounds and the Z-index of sibling views always takes precedence if a touch hits two overlapping views.") but it makes it much easier to make UIs that match mocks while supporting high accessibility standards. I'm unsure how viable it would be to add something like this to flutter, but it would be extremely useful.

**Comment by goderbauer** [comment]
Related issue: https://github.com/flutter/flutter/issues/615

**Comment by Hixie** [comment]
You can set the behaviour of a gesture detector to go all the way to its bounds rather than using its child's bounds, which is basically equivalent to increasing the "hit slop" to maximum given the definition above.

I'm not really sure what we can do beyond that, though...

**Comment by zzzev** [comment]
The raw functionality you describe is a good start, but maybe not very usable in practice. Maybe Flutter could provide a hitSlop-like property on GestureDetector, which if provided would wrap the GestureDetector in a container that attempts to size itself appropriately and handle taps within a certain distance of its child? Then, InkWell, MaterialButton, etc. could also have this property and pass it through to the underlying GestureDetector. Otherwise we'd have to reimplement a lot of the Widgets that Flutter provides to do this.

**Comment by Hixie** [comment]
The GestureDetector already tries to be as big as possible.

**Comment by goderbauer** [comment]
Related: https://github.com/flutter/flutter/issues/16625

**Comment by Hixie** [comment]
This specific request — being able to make a descendant have a bigger hit test region than an ancestor — is not something we will implement, as it is counter to the entire design of the hit testing feature, which was designed specifically with performance in mind; implementing this would likely require a design that compromises on performance.

We can still fix specific instances of the underlying issues such as https://github.com/flutter/flutter/issues/615 and 
https://github.com/flutter/flutter/issues/16625 by making the actual widgets bigger than their rendering.

In conclusion, I'm going to close this bug, but please file widget-specific bugs if there are other specific widgets that need a bigger tap target area.

**Comment by zzzev** [comment]
To clarify: I was not requesting that children have bigger hit test regions than their ancestors; a version of this with identical  limitations as the RN version would still be useful (and those limitations also forbid what you describe). What do you think about the proposal outlined [in this comment](https://github.com/flutter/flutter/issues/14794#issuecomment-380613970)?

**Comment by lukepighetti** [comment]
@Hixie How would you recommend we use a `Padding` widget to increase the size of a hitbox? I have also run into this problem and in RN each view has a `hitSlop` property that allows you to adjust the hitbox of any view.

<img width="800" alt="screen shot 2018-06-17 at 8 56 34 pm" src="https://user-images.githubusercontent.com/666539/41514029-96c89a06-7271-11e8-9044-58391d1d67b4.png">

**Comment by lukepighetti** [comment]
Padding in my experience is unacceptable way to adjust the hitbox because it adjusts the items on the screen.

<img width="307" alt="screen shot 2018-06-17 at 9 01 52 pm" src="https://user-images.githubusercontent.com/666539/41514067-f1dbc7d8-7271-11e8-9649-15e20ac5b30b.png">

<img width="300" alt="screen shot 2018-06-17 at 9 03 33 pm" src="https://user-images.githubusercontent.com/666539/41514068-f1ec2074-7271-11e8-8b8c-a169a947928e.png">

**Comment by lukepighetti** [comment]
@Hixie I know you closed this but I'm being a rebel and opening a new Issue because I believe this to be critical.
